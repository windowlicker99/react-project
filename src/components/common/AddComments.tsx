import React, { ChangeEventHandler, FC, useState } from 'react';
import { getIn, useFormikContext } from 'formik';
import { Textarea } from '@components/common/form-controls/Textarea';
import { Btn } from '@components/common/form-controls/Btn';

interface AddCommentProps {
  name: string;
}

export const AddComments: FC<AddCommentProps> = ({ name }) => {
  const { errors, touched, setFieldValue } = useFormikContext();
  const [comment, setComment] = useState('');
  const handleChangeFieldValue: ChangeEventHandler<HTMLTextAreaElement> = (e) => {
    const {
      target: { value },
    } = e;
    setComment(value);
  };
  const handleResetComment = () => setComment('');
  const handleApplyComment = () => setFieldValue(name, comment);
  return (
    <>
      <Textarea
        placeholder="Add a message"
        error={getIn(touched, name) && getIn(errors, name)}
        value={comment}
        name={name}
        onChange={handleChangeFieldValue}
      />
      <div className="btn-group mt-30 flex-end">
        <Btn btnStatus="btn-lg-outline mr-15" btnText="Cancel" onClick={handleResetComment} />
        <Btn btnStatus="btn-lg-primary" btnText="Apply" disabledBtn={!comment} onClick={handleApplyComment} />
      </div>
    </>
  );
};
