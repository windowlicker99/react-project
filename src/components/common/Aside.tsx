import React, { FC, ReactNode } from 'react';
import classNames from 'classnames';

interface IAsideProps {
  header?: ReactNode;
  asideStatus?: string;
  footer?: ReactNode;
}

export const Aside: FC<IAsideProps> = ({ asideStatus = '', header, footer, children }) => (
  <div className={classNames('aside', asideStatus)}>
    <div className="aside-header">{header}</div>
    <div className="aside-body">{children}</div>
    <div className="aside-footer"> {footer}</div>
  </div>
);
