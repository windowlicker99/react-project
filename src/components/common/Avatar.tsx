import React, { FC } from 'react';
import AvatarImg from '@images/avatar.jpeg';

interface IAvatarProps {
  name: string;
  avatar: string;
}

export const Avatar: FC<IAvatarProps> = ({ name, avatar }) => (
  <div className="avatar-card">
    <div className="avatar">
      <img className="avatar-img" src={avatar || AvatarImg} alt="avatar" />
    </div>
    <span className="avatar-name">{name || 'John Stone'}</span>
  </div>
);
