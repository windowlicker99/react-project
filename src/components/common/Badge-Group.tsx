import React, { FC } from 'react';
import { EBadgeText } from '@interfaces/enums';
import { Badge } from './Badge';

interface IProps {
  type: string;
}
export const BadgeGroup: FC<IProps> = ({ type }) => (
  <div className="badge-group">
    <span className="badge-group-labe">Risk Level</span>
    <Badge badgeType={type} badgeText={EBadgeText[type as keyof typeof EBadgeText]} />
  </div>
);
