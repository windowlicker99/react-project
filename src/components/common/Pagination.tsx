import React, { FC } from 'react';
import { useSelector } from 'react-redux';
import classNames from 'classnames';
import {
  nextPage,
  prevPage,
  pageAmount,
  currentPage,
  pickPage,
  disableNext,
  disablePrev,
} from '@store/selectors/pagination';

export const Pagination: FC = () => {
  const setNextPage = useSelector(nextPage);
  const setPrevPage = useSelector(prevPage);
  const setPageAmount = useSelector(pageAmount);
  const selectedPage = useSelector(currentPage);
  const goToPage = useSelector(pickPage);
  const disableNextIcon = useSelector(disableNext);
  const disablePrevIcon = useSelector(disablePrev);

  const pagesArray = Array.from(Array(setPageAmount).keys());

  return (
    <nav className="pagination" aria-label="pagination">
      <ul className="pagination-list">
        <li className="page-item disabled" role="presentation" onClick={setPrevPage}>
          {disablePrevIcon && (
            <span className="page-link prev">
              <span className="icon-prev" />
            </span>
          )}
        </li>

        {pagesArray.map((page) => (
          <li className={classNames('page-item', { active: selectedPage === page })} key={page}>
            <span className="page-link" role="presentation" onClick={() => goToPage(page)}>
              {page + 1}
            </span>
          </li>
        ))}

        <li className="page-item" role="presentation" onClick={setNextPage}>
          {disableNextIcon && (
            <span className="page-link next">
              <span className="page-link next" />
            </span>
          )}
        </li>
      </ul>
    </nav>
  );
};
