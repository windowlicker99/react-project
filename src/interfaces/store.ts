import { ETableTypes, IColumn, ICurrentElement, IElementsTabsColumnsList, IUser } from '@interfaces/interfaces';

export interface IAuth {
  user: IUser;
}
export interface IPagination {
  nextPage: () => void;
  prevPage: () => void;
  pickPage: (indexPage: number) => void;
  disablePrev: boolean;
  disableNext: boolean;
  currentPage: number;
  pageAmount: number;
  currentTab: string;
  totalElement: number;
  totalElementOnPage: number;
  showedElements: number;
}

export interface IVehicleStore {
  currentVehicle: string;
}

export interface IHeaderStore {
  subtitles: string[];
}

export interface IChangeTypeAction {
  type: ETableTypes;
}

export interface IUpdateDisplayedColumnsAction {
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnState {
  type: ETableTypes;
  columns: IElementsTabsColumnsList;
}

export interface IStoreState {
  auth: IAuth;
  pagination: IPagination;
  vehicles: IVehicleStore;
  customizeColumns: ICustomizeColumnState;
  header: IHeaderStore;
  element: IElementStore;
}

export interface IDisplayColumnsList {
  [key: string]: boolean;
}

export interface ICustomizeColumnAction {
  type: ETableTypes;
  displayColumnsList: IDisplayColumnsList;
}

export interface ICustomizeColumnsState {
  [key: string]: IColumn[];
}

export interface IElementStore {
  currentElement: ICurrentElement;
}
