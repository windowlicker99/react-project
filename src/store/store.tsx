import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import { authReducer } from '@store/slices/authSlice';
import { columnsReducer } from '@store/slices/columnsSlice';
import { paginationReducer } from '@store/slices/paginationSlice';
import { vehiclesReducer } from '@store/slices/vehicleSlice';
import { headerReducer } from '@store/slices/headerSlice';
import { elementReducer } from '@store/slices/elementSlice';

const reducers = combineReducers({
  auth: authReducer,
  vehicles: vehiclesReducer,
  customizeColumns: columnsReducer,
  pagination: paginationReducer,
  header: headerReducer,
  element: elementReducer,
});

const persistConfig = {
  key: 'root',
  storage,
  whiteList: ['auth', 'customizeColumns'],
};

const persistedReducer = persistReducer(persistConfig, reducers);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: false,
    }),
  ],
});

export const persistor = persistStore(store);
