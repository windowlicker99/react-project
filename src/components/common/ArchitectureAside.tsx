import React, { FC } from 'react';
import { Heading } from '@components/common/Heading';
import { Aside } from './Aside';
import { ICurrentElement } from '@/interfaces/interfaces';

interface IArchitectureAsideProps {
  asideStatus?: string;
  currentElement: ICurrentElement;
}

export const ArchitectureAside: FC<IArchitectureAsideProps> = ({ asideStatus = '', children, currentElement }) => (
  <Aside
    asideStatus={asideStatus}
    header={
      currentElement ? (
        <Heading headingText="Details" linkText="Edit details" linkIcon="icon-ellipsis-vertival" />
      ) : null
    }
  >
    {children}
  </Aside>
);
