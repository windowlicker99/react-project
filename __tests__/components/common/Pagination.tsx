import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '@store/store';
import { Pagination } from '@components/common/Pagination';

describe('Pagination tests', () => {
  it('Should render', () => {
    const { container } = render(
      <Provider store={store}>
        <Pagination />
      </Provider>
    );
    expect(container.getElementsByClassName('pagination').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Pagination />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
