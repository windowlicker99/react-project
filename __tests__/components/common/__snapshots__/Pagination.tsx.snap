// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination tests Should match snapshot 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="page-item disabled"
        role="presentation"
      />
      <li
        class="page-item"
        role="presentation"
      >
        <span
          class="page-link next"
        >
          <span
            class="page-link next"
          />
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
